internal/internalapi/internalapi.go

```diff
@@ -81,3 +83,39 @@ func ParseRequestHeaderLabelMapping(s string) (map[string]string, error) {
 
 	return result, nil
 }
+
+// ParseExtraEnvVars parses semicolon-separated key=value pairs into a list of
+// environment variables. The input delimiter is a semicolon (';') to allow
+// values to contain commas without escaping.
+// Example: "OTEL_SERVICE_NAME=ai-gateway;OTEL_TRACES_EXPORTER=otlp".
+func ParseExtraEnvVars(s string) ([]corev1.EnvVar, error) {
```

Comment:
internalapi is intended for sharing logic among multiple components (extproc<>controller, etc.). I think this can live within controller package without exporting it?